mathesar_ui/src/stores/table-data/columns.ts

```diff
@@ -67,7 +68,7 @@ function api(url: string) {
   };
 }
 
-export class ColumnsDataStore implements Writable<ColumnsData> {
+export class ColumnsDataStore extends EventHandler implements Writable<ColumnsData> {
```

Comment:
Small note...

Generally I lean towards using composition over inheritance as much as possible. So for a situation like this I'd be inclined to add an `eventHandler` property on `ColumnsDataStore` instead of having `ColumnsDataStore` inherit from `EventHandler`.  That said, I recognize there are situations where inheritance is a better fit. I don't know this code deeply enough to have a well-formed opinion of whether inheritance is best for this case but just wanted to mention this opinion generally. We could always refactor this to use composition later on if needed.

Reply:
> Generally I lean towards using composition over inheritance as much as possible

This is also my preferred approach when there is a clear separation of concerns.

> I'd be inclined to add an `eventHandler` property on `ColumnsDataStore`

Composition would not be a good idea in this particular case.

If we have an `eventHandler` property, we would have to expose it. So we'd do `columnsDataStore.eventHandler.on('event', value);`. However, this also leads to exposing properties we wouldn't want to expose, such as `eventHandler.dispatch` and `eventHandler.destroy`.

The other possible way is to make `eventHandler` private, and have a wrapper method for `eventHander.on` on the `ColumnsDataStore` class. This would be boilerplate code we'd have to add in all the classes have need an event handler.

Inheritance works perfectly for such situations. We can decide this on a case by case basis and I don't think we should always lean towards a particular approach.